docs/refs/api/rtos/callbacks.md

```diff
@@ -37,96 +19,15 @@ Serial serial(USBTX, USBRX);
  }
 ```
 
-#### The importance of state (a.k.a why not function pointers?)
-
-Callbacks have two important pieces of information, the code to execute, and the state associated with the callback.
-
-A common API design mistake is to use a callback type that doesn’t allow a user to attach state to a callback. The most common example of this is a simple C function pointer:
-
-
-```c++
-class ADC {
-public:
-    // Here, the adc_callback_t type is a function that takes in data
-    typedef void (*adc_callback_t)(float data);
-
-
-    // In this example, the ADC read function calls the user-provided callback
-    // when data is available.
-    void attach(adc_callback_t cb);
-};
-```
+#### The importance of state
 
-This API is sufficient for simple applications, but falls apart when there are multiple ADC modules available. This problem becomes especially noticeable when a user tries to reuse the same procedure for multiple callbacks.
+Callbacks have two important pieces of information, the code to execute and the state associated with the callback.
 
-For example, consider applying a low-pass filter to two different ADC modules
+Without state, callbacks offer limited composability. In C, you can fix this by adding an additional “state” argument to the function pointer, so you can pass in the opaque “state” when you register the callback.
 
-``` c++
-// Here is a small running-average low-pass filter.
-float low_pass_result;
-void low_pass_step(float data) {
-     low_pass_result = low_pass_result*0.99 + data*0.01;
-}
+One of the core features of C++ is the encapsulation of this “state” in classes, with operations that modify the state being represented as member functions in the class. Unfortunately, member function pointers are not compatible with standard function pointers, but you can use member function pointers to pass in state as a C++ object.
```

Comment:
I'm worried comparing the terms "standard function pointers" and "member function pointers" is just going to be confusing if we don't establish what they are.